### PR TITLE
T1825 WIP

### PR DESCRIPTION
**Pull request by humphd**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/49182e387155671/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/49182e387155671/output.txt

Total script time: 0.76 mins

Server is running on http://pr705.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/758781c9bf5fa97/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/758781c9bf5fa97/output.txt

Total script time: 0.95 mins

Server is running on http://pr705.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/54b723be24927c0/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/54b723be24927c0/output.txt

Total script time: 1.53 mins

Server is running on http://pr705.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/45de3c229b50f99/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/45de3c229b50f99/output.txt

Total script time: 0.25 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/ff46f0b4521b422/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/ff46f0b4521b422/output.txt

Total script time: 0.60 mins

Server is running on http://pr705.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/cb4f603b95abb95/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/cb4f603b95abb95/output.txt

Total script time: 0.71 mins

Server is running on http://pr705.butyr.org/templates/test.html

File: cornfield/views/embed.jade [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  head
+    !{templateScripts}
+    !{externalAssets}
+    !{popcorn}
+    !{customData}
+    link(rel='stylesheet', href='/css/embed.css')
+    // This is the embed, the actual content is at the following URL
+    link(rel='cannonical', href='/#{id}.html')
+    style
+      @-o-viewport { width: device-width; }
```
Comment (k88hudson) [inline cornfield/views/embed.jade:12]: Any reason why this is this not in the stylesheet?

File: src/embed.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+  function cancelFullscreen() {
+    // Prefix + case differences.
+    document.exitFullscreen && document.exitFullscreen();
+    document.mozCancelFullscreen && document.mozCancelFullscreen();
+    document.mozCancelFullScreen && document.mozCancelFullScreen();
+    document.webkitCancelFullscreen && document.webkitCancelFullscreen();
+  }
+
+  function $( id ) {
```
Comment (k88hudson) [inline src/embed.js:23]: Any way this could conflict with jQuery at some point if it's included in export?

File: src/embed.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    return document.getElementById( id );
+  };
+
+  function isVisible( id ) {
+    return $( id ).style.display !== "none";
+  }
+
+  function show( elem ) {
+    elem = $( elem );
+    elem.style.display = "block";
```
Comment (k88hudson) [inline src/embed.js:36]: Why not just set `elem.style.display  = ""`

File: cornfield/views/embed.jade (hunk 1 above)
Comment (humphd) [inline cornfield/views/embed.jade:12]: I'm literally copying what YouTube does, and I'm not sure why they do it this way, but they do.  We can move into the embed.css stylesheet if it won't change anything.  What do you want me to do?

File: src/embed.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        cancelFullscreen();
+      } else {
+        requestFullscreen( $( "container" ) );
+      }
+    }, false );
+  }
+
+  function setupEventHandlers( popcorn, config ) {
+    popcorn.on( "ended", function() {
+      console.log( "ended" );
```
Comment (k88hudson) [inline src/embed.js]: Remove

File: src/embed.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+      show( "post-roll" );
+    });
+
+    popcorn.on( "pause", function() {
+      $( "play-pause" ).innerHTML = "Play";
+    });
+
+    popcorn.on( "playing", function() {
+      console.log( "playing" );
```
Comment (k88hudson) [inline src/embed.js]: Did you leave these in for a reason?

File: src/embed.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+      popcorn.play();
+    }
+  }
+
+  var req = requirejs.config({
+    context: "embed",
+    baseUrl: "/src" // ../"
+    // Paths are relative to the baseUrl
+//    paths: {
+//      'text': '../external/require/text'
```
Comment (k88hudson) [inline src/embed.js]: double quotes

File: cornfield/views/embed.jade (hunk 1 above)
Comment (k88hudson) [inline cornfield/views/embed.jade:12]: Yeah, just move this to the top of the stylesheet you include

File: src/embed.js (hunk 2 above)
Comment (humphd) [inline src/embed.js:23]: No, since I don't use jquery internally.

File: src/embed.js (hunk 3 above)
Comment (humphd) [inline src/embed.js:36]: That's bad practice, since it needs to be one of block, inline, none, etc.

**Pull request merged**